src/Razor/test/Microsoft.VisualStudio.LanguageServerClient.Razor.Test/HtmlCSharp/DefinitionHandlerTest.cs

```diff
@@ -0,0 +1,372 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Threading;
+using Moq;
+using Xunit;
+using Range = Microsoft.VisualStudio.LanguageServer.Protocol.Range;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor.HtmlCSharp
+{
+    public class DefinitionHandlerTest
```

Comment:
```suggestion
    public class GoToDefinitionHandlerTest
```

Reply:
In general really good test coverage. You captured pretty much every interesting scenario that we'd want to ensure worked. Only suggestion I'd have is to see if you could build a few helper methods to clean up the bodies of the various test methods since there's a lot of boiler plate.

Reply:
I tried but every test case is subtly different in how we setup mocks. So the ROI on doing this is so low. But I will keep this in mind as I send other PRs like `GoToImplementation` and if I find a way to share code between them I will.